### PR TITLE
[DOI-1456] Add user related accounts endpoint

### PR DESCRIPTION
**Pull request by fgchaio**
This new endpoint will return UserAccounts related to a user. The only typeId we will be using for now is 3 (colaborator) but in the future we might have other user types.

### CODE REVIEW

File: Doppler.UsersApi/Model/BasicUserInformation.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+using System;
+
+namespace Doppler.UsersApi.Model
+{
+    public class BasicUserInformation
+    {
+        public string Email { get; set; }
+        public string Firstname { get; set; }
+        public string Lastname { get; set; }
+        public DateTime InvitationDate { get; set; }
```
Comment (RodrigoPereyraDiaz) [inline Doppler.UsersApi/Model/BasicUserInformation.cs:10]: All users will have an invitation?
Or could be `null` when is registered by own without invitation?

File: Doppler.UsersApi/Infrastructure/IAccountRepository.cs [hunk 2]
```diff
@@ -8,5 +9,6 @@ public interface IAccountRepository
     {
         Task<ContactInformation> GetContactInformation(string accountName);
         Task UpdateContactInformation(string accountName, ContactInformation contactInformation);
+        Task<List<BasicUserInformation>> GetRelatedUsers(string email, int userType);
```
Comment (RodrigoPereyraDiaz) [inline Doppler.UsersApi/Infrastructure/IAccountRepository.cs:12]: Could be nice to add the parameter for `CancellationToken` propagation with a `default` value

```suggestion
        Task<List<BasicUserInformation>> GetRelatedUsers(string email, int userType, CancellationToken cancellationToken = default);
```

File: Doppler.UsersApi/Model/BasicUserInformation.cs (hunk 1 above)
Comment (fgchaio) [reply Doppler.UsersApi/Model/BasicUserInformation.cs:10]: You are right! I thought about it and forgot to allow that scenario.
For now every user returned will have an invitation, but in the future depending on the user type they might not.

**Comment by fgchaio** [comment]
We might need to change the query based on whats resolved on [this PR](https://github.com/MakingSense/Doppler/pull/10558). 
I will wait for resolution before merging

**Pull request merged**